From 48b3c243264c176edb5bac88f71e999ee727f286 Mon Sep 17 00:00:00 2001
From: Jack Todaro <jack.todaro@posteo.net>
Date: Thu, 6 Feb 2020 18:26:08 +1100
Subject: [PATCH] Portage/PackageId.hs: use parseCabalPackageName instead of
 parsec

Just as in a previous commit, we need to use parseCabalPackageName
to parse the Cabal PackageName type, since the 'parsec' method
immediately fails on input of a version number.

Given that we frequently parse package names with their version appended,
we need to use our in-house parser which backtracks upon reaching
a version number. In the case of the bug referenced below, this caused
issues with parsing packages available in the Portage tree (::gentoo)
and the haskell overlay, which is used in the 'status' and 'list' commands.

Bug: https://github.com/gentoo-haskell/hackport/issues/63
Reported-By: Philippe Baril Lecavalier

Signed-off-by: Jack Todaro <jack.todaro@posteo.net>
---
 Portage/PackageId.hs | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Portage/PackageId.hs b/Portage/PackageId.hs
index 9496ee52a..18289c5a4 100644
--- a/Portage/PackageId.hs
+++ b/Portage/PackageId.hs
@@ -61,7 +61,7 @@ instance Parsec PackageName where
   parsec = do
     category <- parsec
     _ <- P.char '/'
-    name <- parsec
+    name <- parseCabalPackageName
     return $ PackageName category name
 
 instance Pretty PackageId where
-- 
2.25.0

